### ISSUE TITLE
MySQL  Status ERROR

### ISSUE THREAD

**Issue by Nirjonadda** [engintron/engintron#430]
I am getting error via MySQL Status. Please let me know the fix. I am using 10.1.21-MariaDB, CENTOS 7.3 x86_64 kvm – WHM 62.0 (build 15) with PHP 7.1.2

> MySQL Status:
 ERROR! MySQL is running but PID file could not be found

http://image.prntscr.com/image/44259847b52a4149b08d90cce50aeb42.png

**Comment by justinbeatz** [comment]
Edit your my.cnf file under MySQL and comment out the line with pid-file option, it could be line 65, then restart mysql.

**Comment by ubidzz** [comment]
It not your server it the command that Engintron is using to get the status for mysql.

Engintron is using this command and it's incorrect for cPanel.
`service mysql status`

The correct command should be using is this.
`service mysqld status`

I don't know if cloudlinux use this command **service mysql status** but i know CentOS does not. I will make a pull request to get this fixed in Engintron for CentOS.

**Comment by Nirjonadda** [comment]
@fevangelou Please let me fix on this issue. Thanks

**Comment by ubidzz** [comment]
@Nirjonadda I put in a pull request and can only hope he changes the command.

You can fix it your self if you want by follow a few step shown below.
SSH in to your server and run the commands shown below
`cd /usr/local/cpanel/whostmgr/docroot/cgi`
`nano engintron.php`

Look for this code!
```
case "mysql_status":
            $command = shell_exec("service mysql status");
```
and replace the **service mysql status** to **service mysqld status**

**Ctrl+X** (To exit)
press **Y** (To save the file)
press Enter (To finish and exit)

Now just reload the Engintron plugin in WHM and your finished.

**Comment by fevangelou** [comment]
What cPanel and EA versions do you have?

**Comment by ubidzz** [comment]
I had the same problem and i have CentOS 7.3, WHM 62 (Build 15) and EA4

**Comment by fevangelou** [comment]
Then this
```service mysqld status```
will produce something like:
```
Redirecting to /bin/systemctl status  mysqld.service
Unit mysqld.service could not be found.
```

On the other hand, "service mysql status" will produce the desired result.

Closing as off-topic.

**Comment by ubidzz** [comment]
No the output is this when you run **service mysqld status** in CentOS 7.3 and i already fixed it in my server and this is the output from the WHM Engintron plugin

MySQL Status:
● mariadb.service - MariaDB database server

   Loaded: loaded (/usr/lib/systemd/system/mariadb.service; enabled; vendor preset: disabled)

  Drop-In: /etc/systemd/system/mariadb.service.d

           └─migrated-from-my.cnf-settings.conf

   Active: active (running) since Fri 2017-02-24 01:25:13 EST; 4h 51min ago

 Main PID: 24632 (mysqld)

   Status: "Taking your SQL requests now..."

   CGroup: /system.slice/mariadb.service

           └─24632 /usr/sbin/mysqld



Feb 24 01:25:13 manage.****** mysqld[24632]: 2017-02-24  1:25:13 140037693061376 [Note] InnoDB: Highest supported file format is Barracuda.

Feb 24 01:25:13 manage.****** mysqld[24632]: 2017-02-24  1:25:13 140037693061376 [Note] InnoDB: 128 rollback segment(s) are active.

Feb 24 01:25:13 manage.****** mysqld[24632]: 2017-02-24  1:25:13 140037693061376 [Note] InnoDB: Waiting for purge to start

Feb 24 01:25:13 manage.****** mysqld[24632]: 2017-02-24  1:25:13 140037693061376 [Note] InnoDB:  Percona XtraDB (http://www.percona.com) 5.6.34-79.1 started; log sequence number 127306517

Feb 24 01:25:13 manage.****** mysqld[24632]: 2017-02-24  1:25:13 140036955670272 [Note] InnoDB: Dumping buffer pool(s) not yet started

Feb 24 01:25:13 manage.****** mysqld[24632]: 2017-02-24  1:25:13 140037693061376 [Note] Plugin 'FEEDBACK' is disabled.

Feb 24 01:25:13 manage.****** mysqld[24632]: 2017-02-24  1:25:13 140037693061376 [Note] Server socket created on IP: '127.0.0.1'.

Feb 24 01:25:13 manage.****** mysqld[24632]: 2017-02-24  1:25:13 140037693061376 [Note] /usr/sbin/mysqld: ready for connections.

Feb 24 01:25:13 manage.****** mysqld[24632]: Version: '10.1.21-MariaDB'  socket: '/var/lib/mysql/mysql.sock'  port: 3306  MariaDB Server

Feb 24 01:25:13 manage.****** systemd[1]: Started MariaDB database server.

**Comment by Nirjonadda** [comment]
@ubidzz Do you have fixed issue MySQL Status ERROR in MariaDB? If so please let me know the steps.

@fevangelou doing Closing as off-topic this issue. When we are still Facing this Issues.

**Comment by ubidzz** [comment]
i posted the fix here https://github.com/engintron/engintron/issues/430#issuecomment-282265420

**Comment by fevangelou** [comment]
@ubidzz This is a temp solution, not a fix.

@Nirjonadda Execute this from a terminal as root user and post back the results:
```whereis mariadb```
and
```whereis mysql```

**Comment by ubidzz** [comment]
@fevangelou how can this be a temp solution when i been running CentOS 7 for about year now it's always been the same command of **service mysqld status** and **service mysqld restart** and the output is shown above and i also changed the command in the Engintron plugin and i get the same output as in SSH.

**Comment by Nirjonadda** [comment]
@ubidzz Your tips is working but Yes its not permanent solution because we need re-do this steps after every update of Engintron.

@fevangelou No output for "whereis mariadb" Please note: I have used replace the service mysql status to service mysqld status from engintron.php

`[root@na ~]# whereis mariadb
mariadb:[root@na ~]# whereis mysql
mysql: /usr/bin/mysql /usr/lib64/mysql /usr/local/lib/mysql.sock /usr/include/mysql /usr/share/mysql /usr/share/man/man1/mysql.1.gz
[root@na ~]#
`

**Comment by 1uhost** [comment]
While Centos 6 used the command 'service mysqld status' ,  Centos 7  uses systemctl, here's some examples:
`systemctl status mysql` ==> check the status
`systemctl stop mysql` ==> stop MySQL
`systemctl restart mysql `==> restart MySQL
`systemctl start mysql` ==> start MySQL

**Comment by ubidzz** [comment]
I think the command for CentOS 6 service mysqld status was added in CentOS 7 because i can use both in CentOS 7

systemctl status mysql
service mysqld status

and both work for me in my server just service mysqld status add this before the mysql status output.
`Redirecting to /bin/systemctl status  mysqld.service`

**Comment by fevangelou** [comment]
@ubidzz It works for MariaDB apparently, not for MySQL.

@Nirjonadda The thing is, there is no easy way to spot if MariaDB is installed. I'll have to look at the cPanel docs for more info, perhaps something from cPanel configs.

**Comment by ubidzz** [comment]
there is a way by using php and peg_match i already did it.

**Comment by fevangelou** [comment]
@ubidzz Thanks

**Comment by ubidzz** [comment]
For the next version please don't use PHP scripting and use a bash script to do all the command and making the SSL conf file to provent the php open_basedir protection problem and people do have this on for a reason. I know this is not the comment for this problem and i am sorry for posting it and you are vary welcome.

**Comment by 1uhost** [comment]
From a CLI this command will give you the MySQL server version
`mysqladmin version`

**Comment by ubidzz** [comment]
@1uhost yes but he needed a way to only extract the SQL name if it was mySQL or MariaDB then do a check to tell witch is being used.

**Comment by ubidzz** [comment]
Something like 
`awk '/MariaDB / > /MySQL /' path to sql file;`

print out witch one is being used.

**Comment by fevangelou** [comment]
Closing for now.

**Comment by fevangelou** [comment]
@ubidzz Since you have MariaDB installed, can you please tell me what the output is for:
```mysqladmin version | grep -i "mariadb"```

**Comment by Nirjonadda** [comment]
@fevangelou  I am also using MariaDB. This is my output from MariaDB install.

````
[root@na ~]# mysqladmin version | grep -i "mariadb"
mysqladmin  Ver 9.1 Distrib 10.1.21-MariaDB, for Linux on x86_64
Copyright (c) 2000, 2016, Oracle, MariaDB Corporation Ab and others.
Server version          10.1.21-MariaDB
[root@na ~]# mysqladmin version
mysqladmin  Ver 9.1 Distrib 10.1.21-MariaDB, for Linux on x86_64
Copyright (c) 2000, 2016, Oracle, MariaDB Corporation Ab and others.

Server version          10.1.21-MariaDB
Protocol version        10
Connection              Localhost via UNIX socket
UNIX socket             /var/lib/mysql/mysql.sock
Uptime:                 2 days 14 hours 6 min 39 sec

Threads: 7  Questions: 45298775  Slow queries: 0  Opens: 2187  Flush tables: 1  Open tables: 1766  Queries per second avg: 202.589
[root@na ~]#
````

**Comment by fevangelou** [comment]
@all Please update to the build I just pushed, includes a check for MySQL/MariaDB.

**Comment by Nirjonadda** [comment]
@fevangelou Still are not fixed with Build 20170228

````
Database Status:

 ERROR! MySQL is running but PID file could not be found
````

**Comment by fevangelou** [comment]
@Nirjonadda Assuming you don't have anything tweaked in Engintron, please update to today's build. Should fix your issue.
